examples/plot_psifci_alg.py

```diff
@@ -0,0 +1,160 @@
+ground_truth_dag = bnlearn.import_DAG("sachs", verbose=False)
+fig = bnlearn.plot(ground_truth_dag)
+
+# .. note::
+#    The Sachs dataset has previously been preprocessed, and the steps are described
+#    in bnlearn, at the web-page https://www.bnlearn.com/research/sachs05/.
+print(df.head())
+print(df.shape)
+
+# %%
+# Preprocess the dataset
+# ----------------------
+# Since the data is one dataframe, we need to process it into a form
+# that is acceptable by dodiscover's :class:`constraint.PsiFCI` algorithm. We
+# will form a list of separate dataframes.
+unique_ints = df["INT"].unique()
+
+# get the list of intervention targets and list of dataframe associated with each intervention
+intervention_targets = [df.columns[idx] for idx in unique_ints]
+data_cols = [col for col in df.columns if col != "INT"]
+data = []
+for interv_idx in unique_ints:
+    _data = df[df["INT"] == interv_idx][data_cols]
+    data.append(_data)
+
+print(len(data), len(intervention_targets))
+# %%
+# Setup constraint-based learner
+# ------------------------------
+# Since we have access to interventional data, the causal discovery algorithm
+# we will use that leverages CI and CD tests to estimate causal constraints
+# is the Psi-FCI algorithm :footcite:`Jaber2020causal`.
+
+# Our dataset is comprised of discrete valued data, so we will utilize the
+# G^2 (Chi-square) CI test.
+ci_estimator = GSquareCITest(data_type="discrete")
+
+# Since our data is entirely discrete, we can also use the G^2 test as our
+# CD test.
+cd_estimator = GSquareCITest(data_type="discrete")
+
+alpha = 0.05
+learner = PsiFCI(
+    ci_estimator=ci_estimator,
+    cd_estimator=cd_estimator,
+    alpha=alpha,
+    max_combinations=10,
+    max_cond_set_size=4,
+    n_jobs=-1,
+)
+
+# create context with information about the interventions
+ctx_builder = make_context(create_using=InterventionalContextBuilder)
+ctx: Context = (
+    ctx_builder.variables(data=data[0]).num_distributions(6).obs_distribution(False).build()
+)
+
+print(ctx.init_graph)
+print(ctx.f_nodes)
+
+# %%
+# Run the learning process
+# ------------------------
+# We have setup our causal context and causal discovery learner, so we will now
+# run the algorithm using the :meth:`constraint.PsiFCI.fit` API, which is similar to scikit-learn's
+# `fit` design. All fitted attributes contain an underscore at the end.
+learner = learner.fit(data, ctx)
+
+# %%
+# Analyze the results
+# ===================
+# Now that we have learned the graph, we will show it here. Note differences and similarities
+# to the ground-truth DAG that is "assumed". Moreover, note that this reproduces Supplementary
+# Figure 8 in :footcite:`Jaber2020causal`.
+est_pag = learner.graph_
+
+print(f"There are {len(est_pag.to_undirected().edges)} edges in the resulting PAG")
+
+# %%
+# Visualize the full graph including the F-node
+dot_graph = draw(est_pag, direction="LR")
+dot_graph.render(outfile="psi_pag_full.png", view=True, cleanup=True)
```

Comment:
Can you stick the graph plot in the PR description?

Reply:
Yes it is there now. It is also rendered in Sphinx: https://output.circle-artifacts.com/output/job/1db998f5-b742-4a0b-82bc-e79363365f6c/artifacts/0/dev/auto_examples/plot_psifci_alg.html#sphx-glr-auto-examples-plot-psifci-alg-py